sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -505,29 +523,48 @@ static void validateKeyPhrases(List<String> expectedKeyPhrases, List<String> act
     }
 
     /**
-     * Helper method to validate the list of named entities.
+     * Helper method to validate the list of categorized entities.
+     *
+     *  @param expectedCategorizedEntityList categorizedEntities returned by the service.
+     * @param actualCategorizedEntityList categorizedEntities returned by the API.
+     */
+    static void validateCategorizedEntities(List<CategorizedEntity> expectedCategorizedEntityList,
+        List<CategorizedEntity> actualCategorizedEntityList) {
+        assertEquals(expectedCategorizedEntityList.size(), actualCategorizedEntityList.size());
+        expectedCategorizedEntityList.sort(Comparator.comparing(CategorizedEntity::getText));
+        actualCategorizedEntityList.sort(Comparator.comparing(CategorizedEntity::getText));
+
+        for (int i = 0; i < expectedCategorizedEntityList.size(); i++) {
+            CategorizedEntity expectedCategorizedEntity = expectedCategorizedEntityList.get(i);
+            CategorizedEntity actualCategorizedEntity = actualCategorizedEntityList.get(i);
+            validateCategorizedEntity(expectedCategorizedEntity, actualCategorizedEntity);
+        }
+    }
+
+    /**
+     * Helper method to validate the list of PII entities.
      *
-     * @param expectedNamedEntityList namedEntities returned by the service.
-     * @param actualNamedEntityList namedEntities returned by the API.
+     * @param expectedPiiEntityList piiEntities returned by the service.
+     * @param actualPiiEntityList piiEntities returned by the API.
      */
-    static void validateNamedEntities(List<NamedEntity> expectedNamedEntityList,
-        List<NamedEntity> actualNamedEntityList) {
-        assertEquals(expectedNamedEntityList.size(), actualNamedEntityList.size());
-        expectedNamedEntityList.sort(Comparator.comparing(NamedEntity::getText));
-        actualNamedEntityList.sort(Comparator.comparing(NamedEntity::getText));
-
-        for (int i = 0; i < expectedNamedEntityList.size(); i++) {
-            NamedEntity expectedNamedEntity = expectedNamedEntityList.get(i);
-            NamedEntity actualNamedEntity = actualNamedEntityList.get(i);
-            validateNamedEntity(expectedNamedEntity, actualNamedEntity);
+    static void validatePiiEntities(List<PiiEntity> expectedPiiEntityList,
```

Comment:
I know we talked about this, but could you remind me why we decided on creating a new Pii type?
So far, it looks like a lot of duplication code, only for the name. No new property, functionality, or anything

@kristapratico

Reply:
My context on this comes from the discussion on Teams: 

> ...that reminds me that users in our study were a little confused that RecognizePiiEntities returned a NamedEntity.  It's worth considering adding a new PiiEntity (or other name) type.  When the healthcare endpoint is up, the shape of the return type will be different for healthcare entities. It's also possible that PiiEntity will evolve differently from NamedEntity

Reply:
Thanks!
It makes me wonder if we should change it now, instead of waiting/asking for more information to the service team.

Also, if we end up going this route, implementation wise, .. should we then just create an `Entity` base class and then `CategoryEntity` and `PiiEntity` inherit from it? and whatever other future Entities? That way we can add new content to it, but the core is the same and for now it will save all the duplicated code

Reply:
I think we are making the change now (at least it seems Java/Python did at this moment). Entity base class makes sense to me.

Reply:
Although, we would have to come up with a better name as `Entity` would be too generic?
How about calling it `NamedEntity` as that is how even the service derives `CategorizedEntity` and `PiiEntity` from.

Reply:
`NamedEntity` sounds good to me :)
@annelo-msft what do you think?

Reply:
The concern with NamedEntity is that it's confusing to have this type without a Name property on it.  I'd ask what we gain by having a base class here.  Is it in the public API?  I feel like I remember Krzysztof pushing back on my introducing type hierarchies where there isn't a strong need for customers to learn the base type.  We don't know what other entities for other domains will necessarily look like yet, and is it possible that they wouldn't need Category and Subcategory?  If they don't, should they not derive from Entity?  (Do we need Subcategory for PiiEntity, for example?)  Correct me if I'm wrong here, but would adding a base class be a breaking change here?  I think it's not (could be wrong!), and if not, we could add it later when we've seen more examples that would derive from it.

Reply:
I wouldn't see the base class as a public API, more an internal to the API to avoid redundancy.

Reply:
to close here, yeah we can always do this later.